packages/openapi-v3/src/enhancers/types.ts

```diff
@@ -13,7 +13,7 @@ import {OASEnhancerBindings} from './keys';
  */
 export interface OASEnhancer {
   name: string;
-  modifySpec(spec: OpenApiSpec): OpenApiSpec;
+  modifySpec(spec: OpenApiSpec): Promise<OpenApiSpec>;
```

Comment:
I think it can be made backward compatible as follows:

```ts
modifySpec(spec: OpenApiSpec): ValueOrPromise<OpenApiSpec>;
```

Existing enhancers can stay as-is.

Reply:
Nice! It avoids the breaking change.